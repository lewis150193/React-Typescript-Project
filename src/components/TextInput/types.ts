import { ChangeEvent } from "react";

type valueType = string | number | Array<any> | undefined;

export interface TextInputProps {
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
  name?: string;
  value: valueType;
  className: string;
  placeHolder: string;
}
