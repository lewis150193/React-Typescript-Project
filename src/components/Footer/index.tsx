import React from 'react';


const styles = {
    backgroundColor: '#607D8B',
    boxShadow: '0 1px 1px 0 rgba(0, 0, 0, 0.12)',
    boxSizing: 'border-box',
    width: '100%',
    color: 'white',
    textAlign: 'center',
    font: 'normal 18px sans-serif',
    padding: '45px',
    marginTop: '80px'
};

export const Footer = () => (
// @ts-ignore
    <div style={styles}>
        <span>Footer</span>
    </div>

)