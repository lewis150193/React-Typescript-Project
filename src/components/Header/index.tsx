import React from 'react';

    const styles = {
        backgroundColor: '#607D8B',
        boxShadow: '0 1px 1px 0 rgba(0, 0, 0, 0.12)',
        boxSizing: 'border-box',
        width: '100%',
        color: 'white',
        textAlign: 'center',
        font: 'normal 18px sans-serif',
        padding: '45px',
        marginTop: '80px'
    }

export const Header = () => (
// @ts-ignore
    <div style={styles}>
        <span>Header</span>
    </div>
);