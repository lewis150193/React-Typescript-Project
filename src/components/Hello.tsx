import * as React from "react";

interface MyProps {
    name: string;
}

interface extendedProps extends  MyProps{
    age: number
}
export const Hello = (props: extendedProps) => <h1>Hello {props.name}my age is {props.age}</h1>;