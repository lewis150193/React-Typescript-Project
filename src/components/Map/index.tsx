import React, { Component } from 'react';
import GoogleMapReact from 'google-map-react';

const AnyReactComponent = ({ text }: any) => <div>{text}</div>;

    interface Props {
        center: {
            lat: number,
            lng: number
        }
        text: string
    }

class SimpleMap extends Component<Props> {
    constructor(props: any) {
        super(props);

    }

    render() {
        return (
            // Important! Always set the container height explicitly
            <div style={{ height: '50vh', width: '100%' }}>
                <GoogleMapReact
                    bootstrapURLKeys={{ key:'AIzaSyAc74MwP4jB-tL0MB_WG8ItOZXwfgyFLac' }}
                    defaultCenter={this.props.center}
                    defaultZoom={11}
                >
                    <AnyReactComponent
                        lat={this.props.center.lat}
                        lng={this.props.center.lng}
                        text={this.props.text}
                    />
                </GoogleMapReact>
            </div>
        );
    }
}

export default SimpleMap;