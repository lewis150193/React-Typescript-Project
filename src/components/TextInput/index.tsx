
import React from 'react';
import {TextInputProps} from "./types";

export const TextBox = ({ onChange, name, value, className, placeHolder}: TextInputProps) => (
        <input className={className} type='input' onChange={onChange} name={name}  value={value} placeholder={placeHolder}/>
    );

