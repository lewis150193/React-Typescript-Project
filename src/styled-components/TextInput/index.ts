import styles from 'styled-components';
import {TextBox} from "../../components/TextInput";

export const StyledTextBox = styles(TextBox)`
        width: 50%;
        padding: 12px 20px;
        margin: 8px 0;
        box-sizing: border-box;
        border: 2px solid #2196F3;
        background-color: #03A9F4;
        color: white;
        flex: 6;
        

`;