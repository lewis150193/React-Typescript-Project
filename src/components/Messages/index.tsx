import React from "react";
import { MessageBoxProps } from "./types";
import { Div } from "../../styled-components/div";

export const MessageBox = ({ date, title, value }: MessageBoxProps) => (
  <Div>
    <p>
      <strong>Title:</strong>
      {title}
    </p>
    <p>
      <strong>Message:</strong>
      {value}
    </p>
    <p>
      <strong>Id:</strong>
      {date}
    </p>
  </Div>
);
