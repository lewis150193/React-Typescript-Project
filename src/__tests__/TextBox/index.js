import {TextBox} from "../../components/TextInput";
import { shallow } from "enzyme/build";
import Enzyme from 'enzyme/build'
import Adapter from 'enzyme-adapter-react-16/build'
import React from "react";
import {CommonWrapper as jest} from "enzyme";

Enzyme.configure({ adapter: new Adapter()});

const myMock = jest.fn();

const Props = {
    onChange: myMock,
    name: 'name',
    value: 'valueType',
    className: 'string',
    placeHolder: 'placeholder'

};

describe('should test TextBox', function () {
    it('should render TextBox', function () {
        const wrapper = shallow(<TextBox {...Props}/>)
        expect(wrapper).toMatchSnapshot();
    });
    it('should simulate TextBox change', function () {
        const wrapper = shallow(<TextBox {...Props}/>)
        const button = wrapper.find('button').simulate('click');
        expect(button).toBeCalledOnce();
    });
});