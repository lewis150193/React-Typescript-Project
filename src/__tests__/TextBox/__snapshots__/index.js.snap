// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test TextBox should render TextBox 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextBox
    className="string"
    name="name"
    onChange={[MockFunction]}
    placeHolder="placeholder"
    value="valueType"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "string",
      "name": "name",
      "onChange": [MockFunction],
      "placeholder": "placeholder",
      "type": "input",
      "value": "valueType",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "string",
        "name": "name",
        "onChange": [MockFunction],
        "placeholder": "placeholder",
        "type": "input",
        "value": "valueType",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
